exercises/custom-set/.meta/gen.go

```diff
@@ -110,72 +125,78 @@ type binaryOp struct {
-var emptyCases = []unaryBoolCase{
-{{range .J.IsEmpty.Cases}}{ // {{.Description}}
-	{{strs .Set | printf "%#v"}},
 	{{.Expected}},
 },
-{{end}}}
+{{- end}}{{end}}{{end}}
+}
+{{end}}{{end}}
 
-// {{.J.Contains.Description}}
-var hasCases = []eleBoolCase{
-{{range .J.Contains.Cases}}{ // {{.Description}}
+{{define "eleOp"}}{{$property := .PropertyType}}{{with .Groups}}
+// {{ .GroupComment $property}}
+var {{$property}}Cases = []eleOpCase{ {{range .}} {{range .Cases}}
+{{if .PropertyMatch $property}} { // {{.Description}}
 	{{strs .Set | printf "%#v"}},
 	{{str .Element | printf "%q"}},
-	{{.Expected}},
+	{{istrs .Expected | printf "%#v"}},
 },
-{{end}}}
-
-// {{.J.Subset.Description}}
-var subsetCases{{template "binaryBool" .J.Subset.Cases}}
-
-// {{.J.Disjoint.Description}}
-var disjointCases{{template "binaryBool" .J.Disjoint.Cases}}
-
-// {{.J.Equal.Description}}
-var eqCases{{template "binaryBool" .J.Equal.Cases}}
+{{- end}}{{end}}{{end}}
+}
+{{end}}{{end}}
 
-// {{.J.Add.Description}}
-var addCases{{template "eleOp" .J.Add.Cases}}
+{{define "binaryBool"}}{{$property := .PropertyType}}{{with .Groups}}
+// {{ .GroupComment $property}}
+var {{$property}}Cases = []binBoolCase{ {{range .}} {{range .Cases}}
+{{if .PropertyMatch $property}} { // {{.Description}}
+	{{strs .Set1 | printf "%#v"}},
+	{{strs .Set2 | printf "%#v"}},
+	{{.Expected}},
+},
+{{- end}}{{end}}{{end}}
+}
+{{end}}{{end}}
 
-// {{.J.Intersection.Description}}
-var intersectionCases{{template "binaryOp" .J.Intersection.Cases}}
+{{define "binaryOp"}}{{$property := .PropertyType}}{{with .Groups}}
+// {{ .GroupComment $property}}
+var {{$property}}Cases = []binOpCase{ {{range .}} {{range .Cases}}
+{{if .PropertyMatch $property}} { // {{.Description}}
+	{{strs .Set1 | printf "%#v"}},
+	{{strs .Set2 | printf "%#v"}},
+	{{istrs .Expected | printf "%#v"}},
+},
+{{- end}}{{end}}{{end}}
+}
+{{end}}{{end}}
 
-// {{.J.Difference.Description}}
-var differenceCases{{template "binaryOp" .J.Difference.Cases}}
+{{/* begin template body */}}
+package stringset
 
-// {{.J.Union.Description}}
-var unionCases{{template "binaryOp" .J.Union.Cases}}
+{{.Header}}
+
+{{/* These template calls utilize a dict helper function in order to pass multiple parameters. */}}
+{{template "unaryBool"  dict "PropertyType" "empty"        "Groups" .J.Groups}}
+{{template "eleBool"    dict "PropertyType" "contains"     "Groups" .J.Groups}}
+{{template "binaryBool" dict "PropertyType" "subset"       "Groups" .J.Groups}}
+{{template "binaryBool" dict "PropertyType" "disjoint"     "Groups" .J.Groups}}
+{{template "binaryBool" dict "PropertyType" "equal"        "Groups" .J.Groups}}
+{{template "eleOp"      dict "PropertyType" "add"          "Groups" .J.Groups}}
+{{template "binaryOp"   dict "PropertyType" "intersection" "Groups" .J.Groups}}
+{{template "binaryOp"   dict "PropertyType" "difference"   "Groups" .J.Groups}}
+{{template "binaryOp"   dict "PropertyType" "union"        "Groups" .J.Groups}}
```

Comment:
Others discovered this solution prior; some googling helped me find it. Looks odd at first, but solves the problem well.